File: app/src/main/java/com/amandariu/tagger/demo/data/source/local/db/TagEntity.java
package com.amandariu.tagger.demo.data.source.local.db;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import com.amandariu.tagger.ITag;

/**
 * Represents a single table row in the tags database.
 *
 * @author amandariu (11/12/17)
 */
@Entity(tableName = "tags")
public class TagEntity {

    @PrimaryKey
    private int id;

    private String label;
    private String color;

    public TagEntity() {
    }

    public TagEntity(ITag tag) {
        this.id = tag.getId();
        this.label = tag.getLabel();
        this.color = tag.getColor();
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }
}

File: app/src/main/java/com/amandariu/tagger/demo/data/source/local/db/TagEntity.kt
package com.amandariu.tagger.demo.data.source.local.db

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import com.amandariu.tagger.ITag

/**
 * Represents a single table row in the tags database.
 *
 * @author Amanda Riu
 */
@Entity(tableName = "tags")
class TagEntity {

    @PrimaryKey
    var id: Int = 0

    var label: String? = null
    var color: String? = null

    constructor()

    constructor(tag: ITag) {
        this.id = tag.id
        this.label = tag.label
        this.color = tag.color
    }
}
